### ISSUE TITLE
Mocks using inherited interface with 'new' modifier on property doesn't work with LINQ syntax

### ISSUE THREAD

**Issue by jeremyharris99** [moq/moq4#239]
I have an interface which extends another interface, but uses the `new` keyword to add a setter to a given property. When mocking the derived interface, the only way to get the new property to return the correct value is using mock.Setup(...).Returns(...). The LINQ syntax doesn't work, and neither does the SetupAllProperties method.

Using Moq 4.2.1409.1722 or 4.2.1507.0118 the example below passes.

Example:

``` xml
<package id="Moq" version="4.2.1510.2205" targetFramework="net45" />
<package id="NUnit" version="2.6.4" targetFramework="net45" />
```

``` C#
public interface Interface1
{
    bool ABoolean { get; }
}
public interface Interface2 : Interface1
{
    new bool ABoolean { get; set; }
}

[TestFixture]
public class TestNewInterface
{
    [Test]
    public void TestTheInterface()
    {
        //This works
        var i1 = Mock.Of<Interface1>(i => i.ABoolean == true);
        Assert.IsTrue(i1.ABoolean);

        //This works, too
        var i2Mock = new Mock<Interface2>();
        i2Mock.Setup(i => i.ABoolean).Returns(true);
        Assert.IsTrue(i2Mock.Object.ABoolean);

        //This fails
        var i2 = Mock.Of<Interface2>(i => i.ABoolean == true);
        Assert.IsTrue(i2.ABoolean);

        //This also fails
        var i2_2 = new Mock<Interface2>();
        i2_2.SetupAllProperties();
        i2_2.Object.ABoolean = true;
        Assert.IsTrue(i2_2.Object.ABoolean);
    }
}
```

**Comment by david-banister** [comment]
This isn't about the new keyword. You can remove the new keyword and it will still fail. The issue is related to the read-only property in the base interface when there's a read/write property in the derived interface. This looks like the same as the issue I just created: #275

**Comment by shwaindog** [comment]
Hi just came across this issue where I also redefine a property to add a setter in a derived interface and when setting up the Mock it sets up the base interface property with the setup value but the derived interface property is left to return null.  Seems the SetupProperty assumes only one property definition exists between an interface and all its base interfaces.  Using Moq 4.7.1

**Comment by stakx** [comment]
Visiting for triage. I'm fairly certain that this is a duplicate of #162.

**Comment by jeremyharris99** [comment]
@stakx Looks like you're right. It does appear to be a duplicate.

**Comment by stakx** [comment]
Fixed in Moq 4.7.58.
